### ISSUE TITLE
Automate VIC UI plugin install during appliance initialization

### ISSUE THREAD

**Issue by andrewtchin** [vmware/vic-product#1432]
**User Statement:**
As a user I want the UI plugin install to be automated.

**Details:**
- This should ideally be done by appliance admin API
- During appliance initialization, run vic-ui-linux equivalent to install the plugin
- Install command requires vSphere admin creds, vCenter target, and vCenter fingerprint
- Alert the user to restart vSphere client services or offer to SSH into vCenter to do it (optional)

https://vmware.invisionapp.com/share/T2D63MPMR

**Comment by andrewtchin** [comment]
Previously detailed in https://github.com/vmware/vic/issues/6890. Copying estimate from there

**Comment by jooskim** [comment]
A sample run of `install.sh` assembles and runs the following command:

```
./vic-ui-darwin install --key com.vmware.vic --target $TARGET --user administrator@vsphere.local --configure-ova --type=VicApplianceVM --version 1.3.1.687 --company VMware --url $PLUGIN_ZIP_URL --force --thumbprint C1:6D:3E:41:A5:89:5E:89:FB:DB:82:2F:32:82:8E:54:53:8B:65:34 --server-thumbprint 83:A9:0E:6A:9D:E7:2A:43:B6:70:98:31:86:4A:7D:BC:0A:BE:8A:92 --name vSphere Integrated Containers-H5Client --summary Plugin for vSphere Integrated Containers-H5Client
```

**Comment by jooskim** [comment]
`uninstall.sh`

```
./vic-ui-darwin remove \
     --key com.vmware.vic \
     --target $TARGET \
     --user administrator@vsphere.local \
     --password $PW \
     --thumbprint C1:6D:3E:41:A5:89:5E:89:FB:DB:82:2F:32:82:8E:54:53:8B:65:34
```

`upgrade.sh`

```
./vic-ui-darwin install \
     --force \
     --key com.vmware.vic \
     --target $TARGET \
     --user administrator@vsphere.local \
     --password $PW \
     --configure-ova \
     --type=VicApplianceVM \
     --version 1.3.1.687 \
     --company VMware \
     --url $PLUGIN_ZIP_URL \
     --thumbprint C1:6D:3E:41:A5:89:5E:89:FB:DB:82:2F:32:82:8E:54:53:8B:65:34 \
     --server-thumbprint 83:A9:0E:6A:9D:E7:2A:43:B6:70:98:31:86:4A:7D:BC:0A:BE:8A:92 \
     --name vSphere Integrated Containers-H5Client \
     --summary Plugin for vSphere Integrated Containers-H5Client
```

**Comment by andrewtchin** [comment]
We should prefer to use the vic-ui go code directly to remove use of the install script

**Comment by morris-jason** [comment]
It seems that 6.5 and 6.7 can both control the vsphere-client service status from REST, something like: POST /appliance/vmon/service/vsphere-client/restart

See:
https://vdc-repo.vmware.com/vmwb-repository/dcr-public/423e512d-dda1-496f-9de3-851c28ca0814/0e3f6e0d-8d05-4f0c-887b-3d75d981bae5/VMware-vSphere-Automation-SDK-REST-6.7.0/docs/apidocs/index.html#PKG_com.vmware.appliance.vmon

**Comment by zjs** [comment]
@wjun: After #1770, #1771, and vmware/vic-ui#489 are complete, the next step for this Epic will be to rebase each of the feature branches on their respective `master` branch.

When doing so, it's very important to be mindful of changes on `master` to files which were deleted on the feature branches. In most cases, the deleted files were moved to other repositories. If we fixed a bug since the feature branches were created, but don't copy any changes over to where the files were moved to, we will reintroduce those bugs!

When each feature branch has been rebased on `master`, post a new PR in each repository to allow for final review and testing.

When the branches are ready to merge, I recommend using the actual "Merge" feature (which will require updating each repository's settings temporarily, as that choice is usually disabled). "Squash & Merge" is a bad choice because it will lose authorship information and make cherry-picking very hard. "Rebase & Merge" would be acceptable, but is less clear and may be harder to cherry-pick than a proper "Merge".

(As an aside, when you do create a merge commit: write a good commit message for it that summarizes the overall change that was made. The information is scattered across various GitHub issues and some of the commit messages on the branch you're merging, but summarizing it in one place makes browsing git history much more useful. [Example](https://github.com/vmware/vic/commit/634e0a57681a71ff6f8bbad477a2cb1597fcc99d).)

**Comment by zjs** [comment]
This should probably be release noted as an enhancement.

**Comment by morris-jason** [comment]
> wjun closed this 17 hours ago

Great job, Jun!

**Comment by stuclem** [comment]
Documented as new feature in https://github.com/vmware/vic-product/releases/tag/v1.4.3#features. Documented fully in https://vmware.github.io/vic-product/assets/files/html/1.4/vic_vsphere_admin/deploy_vic_appliance.html. 

Removing doc impact labels.
